processor/tailsamplingprocessor/internal/sampling/ottl.go

```diff
@@ -0,0 +1,124 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sampling // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor/internal/sampling"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/expr"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspan"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspanevent"
+)
+
+type ottlConditionFilter struct {
+	sampleSpanExpr      expr.BoolExpr[ottlspan.TransformContext]
+	sampleSpanEventExpr expr.BoolExpr[ottlspanevent.TransformContext]
+	ErrorMode           ottl.ErrorMode
+	logger              *zap.Logger
+}
+
+var _ PolicyEvaluator = (*ottlConditionFilter)(nil)
+
+// NewOTTLConditionFilter looks at the trace data and returns a corresponding SamplingDecision.
+func NewOTTLConditionFilter(settings component.TelemetrySettings, spanConditions, spanEventConditions []string, errMode ottl.ErrorMode) (PolicyEvaluator, error) {
+	filter := &ottlConditionFilter{
+		ErrorMode: errMode,
+		logger:    settings.Logger,
+	}
+
+	var err error
+	if len(spanConditions) > 0 {
+		if filter.sampleSpanExpr, err = filterottl.NewBoolExprForSpan(spanConditions, filterottl.StandardSpanFuncs(), errMode, settings); err != nil {
+			return nil, err
+		}
+	}
+
+	if len(spanEventConditions) > 0 {
+		if filter.sampleSpanEventExpr, err = filterottl.NewBoolExprForSpanEvent(spanEventConditions, filterottl.StandardSpanEventFuncs(), errMode, settings); err != nil {
+			return nil, err
+		}
+	}
+
+	return filter, nil
+}
+
+func (ocf *ottlConditionFilter) Evaluate(ctx context.Context, _ pcommon.TraceID, trace *TraceData) (Decision, error) {
+	ocf.logger.Debug("Evaluating with OTTL conditions filter")
```

Comment:
Is there a way to make this message more useful? Like, by adding the trace ID to it? If the purpose is to give the user feedback that this works, you may want a metric counter instead.

Reply:
I prefer to add a `traceID` to it. 

Actually the metric counter would be better. However, we have this debug log for all other tail sampling policies. I think it's bad when people see the log from other policies but empty output for ottl. They may wonder if it's not working.

And it's under `Debug` log level which may help a very little bit. I think we should either replace all of them with a metric, or keep the log and add more information to make it useful. WDYT?

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/be4b4efbb2dd32c9b1c7ded8720979fdeac33cd2/processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go#L41